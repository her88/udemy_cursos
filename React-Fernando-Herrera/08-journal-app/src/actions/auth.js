import { types } from "../types/types";

//esto es lo mismo que abajo, al encerrar con parentesis el objeto, le decimos que lo estamos retornando
export const login = (uid, displayName) => ({
    type: types.login,
    payload: {
        uid,
        displayName
    }

});


/*export const login = (uid, displayName) => {
    return {
        type: types.login,
        payload: {
            uid,
            displayName
        }
    }
}*/