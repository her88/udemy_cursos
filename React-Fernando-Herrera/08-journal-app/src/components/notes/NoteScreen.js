import React from 'react'
import { NotesAppBar } from './NotesAppBar'

export const NoteScreen = () => {
    return (
        <div className="notes__main-content">

            <NotesAppBar />

            <div className="notes__content">

                <input type="text" placeholder="Some awesome title" className="notes__title-input" autoComplete="off"/>

                <textarea placeholder="What happend today" className="notes__textarea">
                    
                </textarea>

                <div className="note__images">
                    <img  
                        src="https://encrypted-tbn0.gstatic.com/images?q=tbn:ANd9GcQxFPqXmWa3ue-WYt2JhvSnydK5UNUYEzBKJQ&usqp=CAU" 
                        alt="imagen"/>

                </div>

            </div>
            
        </div>
    )
}
