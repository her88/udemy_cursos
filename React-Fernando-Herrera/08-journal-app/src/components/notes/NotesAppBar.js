import React from 'react'
import { useDispatch, useSelector } from 'react-redux';
import moment from 'moment';
import { startSaveNote, startUploading } from '../../actions/notes';

export const NotesAppBar = () => {

    const dispatch = useDispatch();

    const { active:note } = useSelector(state => state.notes);

    const noteDate = moment(note.date);

    const handleUpdateNote = () => {
        dispatch(startSaveNote(note));
    }

    const handlePicture = () => {
       document.querySelector('#fileSelector').click();
    }

    const handleFileChange = (e) => {
        const file = e.target.files[0];
        if( file ) {
            dispatch( startUploading(file));
        }
    }

    return (
        <div className="notes_appbar">
            <span>{`${noteDate.format('D')} ${noteDate.format('MMMM')} ${noteDate.format('YYYY')} `}</span>

            <input 
                id="fileSelector"
                type="file"
                name="file"
                style={{display: 'none'}}
                onChange={ handleFileChange }
            />

            <div>
                <button className="btn" onClick={ handlePicture }>
                    Picture
                </button>

                <button className="btn" onClick={ handleUpdateNote }>
                    Save
                </button>
            </div>
        </div>
    )
}
