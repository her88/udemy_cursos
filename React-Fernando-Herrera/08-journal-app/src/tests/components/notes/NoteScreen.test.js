import React from "react";
import { mount } from 'enzyme';
import configureStore from "redux-mock-store";
import thunk from "redux-thunk";
import { Provider } from "react-redux";
import { MemoryRouter } from "react-router-dom";
import { NoteScreen } from "../../../components/notes/NoteScreen";
import { activeNote } from "../../../actions/notes";

jest.mock('../../../actions/notes', () => ({
    activeNote: jest.fn()
}));

const middlewares = [thunk];
const mockStore = configureStore(middlewares);

const initState = {
    auth: {
        uid: '19a58YaV293XESUt1EZxJfW0Rzpy2',
        name: 'pepe'
    },
    ui: {
        loading: false,
        msgError: null
    },
    notes: {
        active: {
            id: 'QNOp40h3shu6OP3slUQx',
            date: 1642955215207,
            title: 'Esta es una nota para Pepe',
            body: 'Hola pepe como estas?',
            url: 'https://res.cloudinary.com/dzk6hkcip/image/upload/v1642955249/ul8d2bcfy6drm16kqvho.jpg'
        },
        notes: []
    }
}

let store = mockStore(initState);
store.dispatch = jest.fn();

const wrapper = mount(
    <Provider store={store}>
        <NoteScreen />
    </Provider>
)

describe('Pruebas en <NoteScreen />', () => {

    test('debe de mostrarse correctamente ', () => {

        expect(wrapper).toMatchSnapshot();

    });

    test('debe de disparar el active note', () => {

        wrapper.find('input[name="title"]').simulate('change', {
            target: {
                name: 'title',
                value: 'Hola de nuevo'
            }
        });

        expect(activeNote).toHaveBeenLastCalledWith(
            'QNOp40h3shu6OP3slUQx',
            {
                body: 'Hola pepe como estas?',
                title: 'Hola de nuevo',
                id: 'QNOp40h3shu6OP3slUQx',
                date: 1642955215207,
                url: 'https://res.cloudinary.com/dzk6hkcip/image/upload/v1642955249/ul8d2bcfy6drm16kqvho.jpg'
            }
        )

    });


})
