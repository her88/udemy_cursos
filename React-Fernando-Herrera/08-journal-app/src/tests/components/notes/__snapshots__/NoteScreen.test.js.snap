// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en <NoteScreen /> debe de mostrarse correctamente  1`] = `
<div
  className="notes__main-content"
>
  <div
    className="notes_appbar"
  >
    <span>
      23 January 2022 
    </span>
    <input
      id="fileSelector"
      name="file"
      onChange={[Function]}
      style={
        Object {
          "display": "none",
        }
      }
      type="file"
    />
    <div>
      <button
        className="btn"
        onClick={[Function]}
      >
        Picture
      </button>
      <button
        className="btn"
        onClick={[Function]}
      >
        Save
      </button>
    </div>
  </div>
  <div
    className="notes__content"
  >
    <input
      autoComplete="off"
      className="notes__title-input"
      name="title"
      onChange={[Function]}
      placeholder="Some awesome title"
      type="text"
      value="Esta es una nota para Pepe"
    />
    <textarea
      className="notes__textarea"
      name="body"
      onChange={[Function]}
      placeholder="What happened today"
      value="Hola pepe como estas?"
    />
    <div
      className="note__images"
    >
      <img
        alt="imagen"
        src="https://res.cloudinary.com/dzk6hkcip/image/upload/v1642955249/ul8d2bcfy6drm16kqvho.jpg"
      />
    </div>
  </div>
  <button
    className="btn btn-danger"
    onClick={[Function]}
  >
    Delete
  </button>
</div>
`;
